hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -257,6 +257,7 @@ private HoodieWriteConfig createMetadataWriteConfig(HoodieWriteConfig writeConfi
             // we will trigger compaction manually, to control the instant times
             .withInlineCompaction(false)
             .withMaxNumDeltaCommitsBeforeCompaction(writeConfig.getMetadataCompactDeltaCommitMax())
+            .withPreserveCommitMetadata(false)
```

Comment:
can you help me understand why we have to set false for metadata writer? if possible, we should keep the compaction logic consistent everywhere. If compaction is **defined as** not to alter commit time and such, then we better not to have special cases for easy maintenance.

Reply:
fixed and reverted it.